src/utils/migration.ts

```diff
@@ -98,59 +124,75 @@ export const _20220527_120000 = (mnemonic: string, walletName: string) => {
+    : generalSettings
+  const newGeneralSettings = merge({}, defaultSettings.general, migratedGeneralSettings)
 
-  if (
-    settings.network.explorerApiHost === 'https://mainnet-backend.alephium.org' ||
-    settings.network.explorerApiHost === 'https://backend-v18.mainnet.alephium.org'
-  ) {
-    migratedSettings.network.explorerApiHost = 'https://backend-v112.mainnet.alephium.org'
-  } else if (
-    settings.network.explorerApiHost === 'https://testnet-backend.alephium.org' ||
-    settings.network.explorerApiHost === 'https://backend-v18.testnet.alephium.org'
-  ) {
-    migratedSettings.network.explorerApiHost = 'https://backend-v112.testnet.alephium.org'
-  }
+  SettingsStorage.store('general', newGeneralSettings)
+}
 
-  if (settings.network.explorerUrl === 'https://explorer-v18.mainnet.alephium.org') {
-    migratedSettings.network.explorerUrl = 'https://explorer.alephium.org'
-  } else if (
-    settings.network.explorerUrl === 'https://testnet.alephium.org' ||
-    settings.network.explorerUrl === 'https://explorer-v18.testnet.alephium.org'
-  ) {
-    migratedSettings.network.explorerUrl = 'https://explorer.testnet.alephium.org'
+export const _v140_networkSettingsMigration = () => {
+  const oldSettings = SettingsStorage.load('network') as NetworkSettings
+
+  const migratedNetworkSettings = {
+    nodeHost:
+      oldSettings.nodeHost === 'https://mainnet-wallet.alephium.org'
+        ? networkPresets.mainnet.nodeHost
+        : oldSettings.nodeHost === 'https://testnet-wallet.alephium.org'
+        ? networkPresets.testnet.nodeHost
+        : oldSettings.nodeHost,
+    explorerApiHost:
+      oldSettings.explorerApiHost === 'https://mainnet-backend.alephium.org'
+        ? networkPresets.mainnet.explorerApiHost
+        : oldSettings.explorerApiHost === 'https://testnet-backend.alephium.org'
+        ? networkPresets.testnet.explorerApiHost
+        : oldSettings.explorerApiHost,
+    explorerUrl:
+      oldSettings.explorerUrl === 'https://testnet.alephium.org'
+        ? networkPresets.testnet.explorerUrl
+        : oldSettings.explorerUrl
   }
 
-  if (
-    settings.network.nodeHost === 'https://mainnet-wallet.alephium.org' ||
-    settings.network.nodeHost === 'https://wallet-v18.mainnet.alephium.org'
-  ) {
-    migratedSettings.network.nodeHost = 'https://wallet-v16.mainnet.alephium.org'
-  } else if (
-    settings.network.nodeHost === 'https://testnet-wallet.alephium.org' ||
-    settings.network.nodeHost === 'https://wallet-v18.testnet.alephium.org'
-  ) {
-    migratedSettings.network.nodeHost = 'https://wallet-v16.testnet.alephium.org'
-  }
+  const newNetworkSettings = merge({}, defaultSettings.network, migratedNetworkSettings)
+  SettingsStorage.store('network', newNetworkSettings)
+}
 
-  const newSettings = merge({}, defaultSettings, migratedSettings)
-  SettingsStorage.storeAll(newSettings)
+export const _v150_networkSettingsMigration = () => {
```

Comment:
Couldn't the 2 network settings migrations be merged into one?

Quick DRYer proposition:

```ts
export const networkSettingsMigration = () => {
  const oldSettings = SettingsStorage.load('network') as NetworkSettings

  const getNewSettings = (networkSettingName: keyof Omit<NetworkSettings, 'networkId'>) => {
    const networksToSwap = ['mainnet', 'testnet'] as Array<'mainnet' | 'testnet'> // this can prob be done better

    let newSettingsValue = ''

    for (const n of networksToSwap) {
      if (newSettingsValue) return

      newSettingsValue = oldURLs[networkSettingName][n].includes(oldSettings[networkSettingName])
        ? networkPresets[n][networkSettingName]
        : ''
    }

    return newSettingsValue || oldSettings[networkSettingName]
  }

  const migratedNetworkSettings = {
    nodeHost: swapgetNewSettingsSettings('nodeHost'),
    explorerApiHost: getNewSettings('explorerApiHost'),
    explorerUrl: getNewSettings('explorerUrl')
  }

  const newNetworkSettings = merge({}, defaultSettings.network, migratedNetworkSettings)
  SettingsStorage.store('network', newNetworkSettings)
}
```

Reply:
Sorry, I forgot to include my `oldURLs ` object, which made my proposition harder to understand.

But I like your proposition too! It's clear and quite DRY <3